agent/engine/docker_task_engine.go

```diff
@@ -791,6 +805,20 @@ func (engine *DockerTaskEngine) pullAndUpdateContainerReference(task *apitask.Ta
 		defer container.SetRegistryAuthCredentials(credentials.IAMRoleCredentials{})
 	}
 
+	// Apply registry auth data from ASM if required
+	if container.ShouldPullWithASMAuth() {
+		if err := task.PopulateASMAuthData(container); err != nil {
+			seelog.Errorf("Task engine [%s]: unable to acquire Docker registry credentials for container [%s]",
+				task.Arn, container.Name)
+			return dockerapi.DockerContainerMetadata{
+				Error: dockerapi.CannotPullECRContainerError{
```

Comment:
Using `CannotPullECRContainerError` seems confusion here, as this is not necessary pulling from ECR, can you create a new error type?

Reply:
yea you're right, i'm going to add some distinct error types here.

Reply:
added a `CannotPullContainerAuthError` type